apps/preview/next/pages/showcases/Gallery/GalleryHorizontalArrowKeyNavigation.tsx

```diff
@@ -0,0 +1,138 @@
+/* eslint-disable react/no-array-index-key */
+/* eslint-disable jsx-a11y/anchor-has-content */
+import { ShowcasePageLayout } from '../../showcases';
+
+// #region source
+import { useState } from 'react';
+import {
+  SfScrollable,
+  SfButton,
+  SfIconChevronLeft,
+  SfIconChevronRight,
+  type SfScrollableOnDragEndData,
+} from '@storefront-ui/react';
+import * as Images from './images';
+import classNames from 'classnames';
+
+const images = [
+  { imageSrc: Images.gallery1Src, alt: 'backpack1' },
+  { imageSrc: Images.gallery2Src, alt: 'backpack2' },
+  { imageSrc: Images.gallery3Src, alt: 'backpack3' },
+  { imageSrc: Images.gallery4Src, alt: 'backpack4' },
+  { imageSrc: Images.gallery5Src, alt: 'backpack5' },
+  { imageSrc: Images.gallery6Src, alt: 'backpack6' },
+  { imageSrc: Images.gallery7Src, alt: 'backpack7' },
+  { imageSrc: Images.gallery8Src, alt: 'backpack8' },
+  { imageSrc: Images.gallery9Src, alt: 'backpack9' },
+  { imageSrc: Images.gallery10Src, alt: 'backpack10' },
+  { imageSrc: Images.gallery11Src, alt: 'backpack11' },
+];
+
+const thumbImages = [
+  { imageSrc: Images.gallery1ThumbSrc, alt: 'backpack1' },
+  { imageSrc: Images.gallery2ThumbSrc, alt: 'backpack2' },
+  { imageSrc: Images.gallery3ThumbSrc, alt: 'backpack3' },
+  { imageSrc: Images.gallery4ThumbSrc, alt: 'backpack4' },
+  { imageSrc: Images.gallery5ThumbSrc, alt: 'backpack5' },
+  { imageSrc: Images.gallery6ThumbSrc, alt: 'backpack6' },
+  { imageSrc: Images.gallery7ThumbSrc, alt: 'backpack7' },
+  { imageSrc: Images.gallery8ThumbSrc, alt: 'backpack8' },
+  { imageSrc: Images.gallery9ThumbSrc, alt: 'backpack9' },
+  { imageSrc: Images.gallery10ThumbSrc, alt: 'backpack10' },
+  { imageSrc: Images.gallery11ThumbSrc, alt: 'backpack11' },
+];
+
+export default function GalleryHorizontal() {
+  const [activeIndex, setActiveIndex] = useState(0);
+
+  const onDragged = (event: SfScrollableOnDragEndData) => {
+    if (event.swipeRight && activeIndex > 0) {
+      setActiveIndex((currentActiveIndex) => currentActiveIndex - 1);
+    } else if (event.swipeLeft && activeIndex < images.length - 1) {
+      setActiveIndex((currentActiveIndex) => currentActiveIndex + 1);
+    }
+  };
+
+  const activeArrowNavigation = (event: React.KeyboardEvent, index: number) => {
+    event.preventDefault();
+    const currentElement = event?.target as HTMLElement;
+    const nextElement = currentElement.nextElementSibling as HTMLElement;
+    const previousElement = currentElement.previousElementSibling as HTMLElement;
+    if ((event.code === 'ArrowRight' || event.code === 'ArrowUp') && index < thumbImages.length - 1) {
+      setActiveIndex(index + 1);
+      nextElement.focus();
```

Comment:
Out of curiosity I checked what would be faster, get all items and target by index or your way, logically I would say your way, but how JS is read by engine might be misleading

And so we are reading from index is faster, when we have read all items, but that would be only accesible if ref would be used on element and then `ref.children` 
![image](https://github.com/vuestorefront/storefront-ui/assets/2566152/d16e1035-f219-4a84-8fb4-1c5eb94b2f38)


So at the end your way is simpler and close enough fast to lab testing, however we dont need to use `useRef` which in total with reading item by index, probably would be slower

Reply:
Nice that you also check such cases :)